edb/server/pgcon/pgcon.pyx

```diff
@@ -2062,12 +2086,17 @@ cdef class PGConnection:
             while True:
                 field_type = self.buffer.read_byte()
                 if field_type == b'P':  # Position
-                    # Internal query
-                    msg_buf.write_byte(b'q')
-                    # Compiled SQL
-                    msg_buf.write_bytestring(action.args[0])
-                    # Internal position
-                    msg_buf.write_byte(b'p')
+                    qu = (action.query_unit.translation_data
+                          if isinstance(action.query_unit,
+                                        compiler.dbstate.SQLQueryUnit)
```

Comment:
`action.query_unit` has type `SQLQueryUnit | None`, so I think just checking `is None` is sufficient here.